programs/util.c

```diff
@@ -386,7 +386,7 @@ static size_t readLineFromFile(char* buf, size_t len, FILE* file)
     assert(!feof(file));
     if ( fgets(buf, (int) len, file) == NULL ) return 0;
     {   size_t linelen = strlen(buf);
-        if (strlen(buf)==0) return 0;
+        if (buf[0]=='\0') return 0;
```

Comment:
I think we can rely on the compiler to make this optimization, so I prefer the original code here since it's more readable.